### PR TITLE
remove redirect from user settings edit

### PR DESCRIPTION
**Pull request by ryanlerch**
previously, when editng the user settings (profile or gpg keys)
after a successful POST, noggin would redirect back to the profile
page, rather than to the settings page like with other settings pages
like OTP. This changes noggin to return to the settings edit form
rather than the profile page

Signed-off-by: Ryan Lerch <rlerch@redhat.com>

### CODE REVIEW

**Comment by abompard** [comment]
I think that on POST success it should still do a redirect, even if it is to the same page. Otherwise when trying to reload the page users will have the browser's warning message about sending the request again.
In my understanding the only situation where POST should not end up in a redirect is when there is an error with the posted data (in which case it should ideally never end up in a redirect but sometimes that makes things very complicated so we do it sometimes).

**Comment by ryanlerch** [comment]
Thanks @abompard have updated with changes

File: noggin/controller/user.py [hunk 1]
```diff
@@ -80,9 +86,8 @@ def user_settings_profile(ipa, username):
                 'fasgitlabusername': form.gitlab.data.lstrip('@'),
                 'fasrhbzemail': form.rhbz_mail.data,
             },
+            "user_settings_profile",
         )
-        if result:
-            return result
```
Comment (abompard) [inline noggin/controller/user.py:87]: If you're not returning the redirect that `_user_mod` returned, then is it still redirecting?

File: noggin/controller/user.py [hunk 2]
```diff
@@ -97,14 +102,13 @@ def user_settings_keys(ipa, username):
...
-        result = _user_mod(
+        _user_mod(
             ipa,
             form,
             username,
             {'ipasshpubkey': form.sshpubkeys.data, 'fasgpgkeyid': form.gpgkeys.data},
+            'user_settings_keys',
         )
-        if result:
-            return result
```
Comment (abompard) [inline noggin/controller/user.py:109]: Same question here.

File: noggin/tests/unit/controller/test_user.py [hunk 3]
```diff
@@ -75,11 +75,12 @@ def test_user_edit(client, logged_in_dummy_user):
...
     """Test posting to the user edit page: /user/<username>/settings/profile/"""
     result = client.post('/user/dummy/settings/profile/', data=POST_CONTENTS)
-    assert_redirects_with_flash(
+    assert_flashed_message(
         result,
         expected_url="/user/dummy/",
-        expected_message="Profile has been succesfully updated.",
+        expected_message="Profile Updated: <a href=\"/user/dummy/\">view your profile</a>",
         expected_category="success",
+        redirect=False,
```
Comment (abompard) [inline noggin/tests/unit/controller/test_user.py]: If the POST is successful, then it should redirect, no?

File: noggin/tests/unit/controller/test_user.py [hunk 4]
```diff
@@ -106,7 +107,7 @@ def test_user_edit_post_no_change(client, logged_in_dummy_user):
     # Do it once
     result = client.post('/user/dummy/settings/profile/', data=POST_CONTENTS)
 
-    assert result.status_code == 302
+    assert result.status_code == 200
```
Comment (abompard) [inline noggin/tests/unit/controller/test_user.py]: Same thing here, if the POST is successful, it should redirect.

File: noggin/tests/unit/controller/test_user.py [hunk 5]
```diff
@@ -142,11 +143,12 @@ def test_user_settings_keys(client, logged_in_dummy_user):
...
     """Test posting to the user edit page: /user/<username>/settings/keys/"""
     result = client.post('/user/dummy/settings/keys/', data=POST_CONTENTS_KEYS)
-    assert_redirects_with_flash(
+    assert_flashed_message(
         result,
         expected_url="/user/dummy/",
-        expected_message="Profile has been succesfully updated.",
+        expected_message="Profile Updated: <a href=\"/user/dummy/\">view your profile</a>",
         expected_category="success",
+        redirect=False,
```
Comment (abompard) [inline noggin/tests/unit/controller/test_user.py]: Same here.

File: noggin/tests/unit/controller/test_user.py [hunk 6]
```diff
@@ -173,7 +175,7 @@ def test_user_settings_keys_post_no_change(client, logged_in_dummy_user):
     # Do it once
     result = client.post('/user/dummy/settings/keys/', data=POST_CONTENTS_KEYS)
 
-    assert result.status_code == 302
+    assert result.status_code == 200
```
Comment (abompard) [inline noggin/tests/unit/controller/test_user.py]: And same here. Unless I'm missing something?

File: noggin/tests/unit/utilities.py [hunk 7]
```diff
@@ -5,17 +5,22 @@
 import pyotp
 
 
-def assert_redirects_with_flash(
-    response, expected_url, expected_message, expected_category
+def assert_flashed_message(
+    response, expected_url, expected_message, expected_category, redirect=True
```
Comment (abompard) [inline noggin/tests/unit/utilities.py]: I haven't seen a case where there should be no redirect.

**Review by abompard — CHANGES_REQUESTED** [review]

**Review by abompard — APPROVED** [review]
LGTM, thanks!

**Pull request merged**